gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/rao/CseValidRaoValidator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ *  This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.cse_valid.app.rao;
+
+import com.farao_community.farao.cse_valid.api.resource.CseValidRequest;
+import com.farao_community.farao.cse_valid.api.resource.ProcessType;
+import com.farao_community.farao.cse_valid.app.FileExporter;
+import com.farao_community.farao.cse_valid.app.FileImporter;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_creation.creator.cse.CseCrac;
+import com.farao_community.farao.data.rao_result_api.OptimizationState;
+import com.farao_community.farao.data.rao_result_api.RaoResult;
+import com.farao_community.farao.rao_runner.api.resource.RaoRequest;
+import com.farao_community.farao.rao_runner.api.resource.RaoResponse;
+import com.farao_community.farao.rao_runner.starter.RaoRunnerClient;
+import com.powsybl.iidm.network.Network;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.time.OffsetDateTime;
+
+/**
+ * @author Oualid Aloui {@literal <oualid.aloui at rte-france.com>}
+ */
+
+@Component
+public class CseValidRaoValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CseValidRaoValidator.class);
+
+    private final FileImporter fileImporter;
+    private final FileExporter fileExporter;
+    private final RaoRunnerClient raoRunnerClient;
+    private int variantCounter = 0;
+
+    public CseValidRaoValidator(FileImporter fileImporter, FileExporter fileExporter, RaoRunnerClient raoRunnerClient) {
+        this.fileImporter = fileImporter;
+        this.fileExporter = fileExporter;
+        this.raoRunnerClient = raoRunnerClient;
+    }
+
+    public boolean isNetworkSecure(Network network, CseValidRequest cseValidRequest, String cracUrl) {
+        ProcessType processType = cseValidRequest.getProcessType();
+        OffsetDateTime processTargetDateTime = cseValidRequest.getTimestamp();
+        String requestId = cseValidRequest.getId();
+
+        String jsonCracUrl = getJsonCracUrl(cseValidRequest, network, cracUrl);
+        String raoParametersURL = fileExporter.saveRaoParameters(processTargetDateTime, processType);
+        String scaledNetworkDirPath = generateScaledNetworkDirPath(network, processTargetDateTime, processType);
+        String scaledNetworkName = network.getNameOrId() + ".xiidm";
+        String networkFiledUrl = fileExporter.saveNetworkInArtifact(network, scaledNetworkDirPath + scaledNetworkName, "", processTargetDateTime, processType);
```

Comment:
I think it would be cleaner to save these files before calling this method and pass the URLs as argument. I would also separate the runRao() part that is time consuming and call an external service from the isSecure() part that is just file reading and a basic test.
Towards the file saving it would be cleaner to separate so that we can use these URLs in the dichotomy (that comes after this computation) without saving them again separately before the dichotomy.